### PR TITLE
[healthcheck] Remove deprecated fields from TabletHealth json marshalling

### PR DESCRIPTION
**Pull request by ajm188**
## Description

Plus, as a bonus, resolve a few TODO comments in the tests.

## Related Issue(s)

N/A

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported (N/A)
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: doc/releasenotes/17_0_0_summary.md [hunk 1]
```diff
@@ -160,3 +161,7 @@ Some notes to help understand these metrics:
  * `DurationByPhaseSeconds["CatchUpReplication"]` measures how long it took to catch-up replication after the restore phase.
  * `DurationByPhaseSeconds["RestoreLastBackup"]` measures to the duration of the restore phase.
  * `RestoreDurationNanoseconds["-.-.Restore"]` also measures to the duration of the restore phase.
+
+## <a id="deprecations-and-deletions"/> Deprecations and Deletions
```
Comment (ajm188) [inline doc/releasenotes/17_0_0_summary.md:165]: fully prepared to accept the consequence i'm going to conflict with my own PR (#12427) here

**Review by GuptaManan100 — APPROVED** [review]
LGTM

File: doc/releasenotes/17_0_0_summary.md (hunk 1 above)
Comment (frouioui) [reply doc/releasenotes/17_0_0_summary.md:174]: The merge conflict above is on purpose?

**Review by frouioui — APPROVED** [review]

**Review by deepthi — APPROVED** [review]
The summary file needs to be fixed up before merging.
Rest LGTM.
Nice work, thank you for doing this!

File: doc/releasenotes/17_0_0_summary.md (hunk 1 above)
Comment (ajm188) [reply doc/releasenotes/17_0_0_summary.md:174]: oh weird, i must have saved before VSCode finished actually doing the resolution. no this wasn't on purpose

**Pull request merged**